packit_service/models.py

```diff
@@ -2042,3 +2046,200 @@ def get_by_dist_git_id(cls, id_: int) -> Optional["SourceGitPRDistGitPRModel"]:
+    task_accepted_time = Column(DateTime)
+    # datetime.utcnow instead of datetime.utcnow() because its an argument to the function
+    # so it will run when the copr build is initiated, not when the table is made
+    build_submitted_time = Column(DateTime, default=datetime.utcnow)
+    build_start_time = Column(DateTime)
+    build_finished_time = Column(DateTime)
+
+    # metadata for the build which didn't make it to schema yet
+    data = Column(JSON)
+
+    runs = relationship("PipelineModel", back_populates="vm_image_build")
+
+    def set_start_time(self, start_time: datetime):
+        with sa_session_transaction() as session:
+            self.build_start_time = start_time
+            session.add(self)
+
+    def set_end_time(self, end_time: datetime):
+        with sa_session_transaction() as session:
+            self.build_finished_time = end_time
+            session.add(self)
+
+    def set_status(self, status: VMImageBuildStatus):
+        with sa_session_transaction() as session:
+            self.status = status
+            session.add(self)
+
+    def set_build_logs_url(self, build_logs: str):
+        with sa_session_transaction() as session:
+            self.build_logs_url = build_logs
+            session.add(self)
+
+    @classmethod
+    def get_by_id(cls, id_: int) -> Optional["VMImageBuildTargetModel"]:
+        return sa_session().query(VMImageBuildTargetModel).filter_by(id=id_).first()
+
+    @classmethod
+    def get_all(cls) -> Iterable["VMImageBuildTargetModel"]:
+        return (
+            sa_session()
+            .query(VMImageBuildTargetModel)
+            .order_by(desc(VMImageBuildTargetModel.id))
+        )
+
+    @classmethod
+    def get_all_by_build_id(
+        cls, build_id: Union[str, int]
+    ) -> Iterable["VMImageBuildTargetModel"]:
+        """Returns all builds with that build_id, irrespective of target"""
+        if isinstance(build_id, int):
+            # See the comment in get_by_build_id()
+            build_id = str(build_id)
+        return sa_session().query(VMImageBuildTargetModel).filter_by(build_id=build_id)
+
+    @classmethod
+    def get_all_by_status(
+        cls, status: VMImageBuildStatus
+    ) -> Iterable["VMImageBuildTargetModel"]:
+        """Returns all builds which currently have the given status."""
+        return sa_session().query(VMImageBuildTargetModel).filter_by(status=status)
+
+    @classmethod
+    def get_by_build_id(
+        cls, build_id: Union[str, int], target: str = None
+    ) -> Optional["VMImageBuildTargetModel"]:
+        """Returns the build matching the build_id and the target"""
+
+        if isinstance(build_id, int):
+            # PG is pesky about this:
+            #   LINE 3: WHERE copr_builds.build_id = 1245767 AND copr_builds.target ...
+            #   HINT:  No operator matches the given name and argument type(s).
+            #   You might need to add explicit type casts.
+            build_id = str(build_id)
```

Comment:
I just ran into this :/ it's odd that copr build_id is str in our db even though it should probably be an int (that's how we work with it in code)